radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/particles/Particle.java

```diff
@@ -1,9 +1,88 @@
 package com.radixdlt.client.core.atoms.particles;
 
+import com.radixdlt.client.core.atoms.particles.quarks.Quark;
 import com.radixdlt.client.core.crypto.ECPublicKey;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Stream;
+
+public abstract class Particle {
+	private final Spin spin;
+	private final List<Quark> quarks;
+
+	protected Particle(Spin spin, Quark... quarks) {
+		this.spin = spin;
+		this.quarks = new ArrayList<>(Arrays.asList(quarks));
```

Comment:
Can you only have one type of quark per particle? If yes, then maybe this should be a map to enforce this constraint. If not, then is getQuark() below broken?

Reply:
Also, I don't believe Arrays.asList needs to be wrapped by ArrayList?

Reply:
@talekhinezh: It depends on how you want to use the resulting `List`. `Arrays.asList(T...)` will produce a fixed-size, immutable `List` implementation (you can't add any further items to it), while the suggested `ArrayList` constructor will not only initialise the resulting `List` implementation with provided items, but it will also allow you to add more, would you like to do so.

I don't know how `this.quarks` is supposed to behave, hence I refrain from recommending either of the mentioned options.

Reply:
Ah, wait a minute! You mean that "fixed-size, immutable" is what we want here, of course! If so, please disregard my previous comment.